src/module-registry.js

```diff
@@ -41,7 +41,10 @@ class ModuleRegistry {
   component(globalID) {
     const generator = this.registeredComponents[globalID];
     if (!generator) {
-      console.error(`ModuleRegistry.component ${globalID} used but not yet registered`);
+      this.notifyListeners('ReactModuleContainerError', {
+        type: 'UnregisteredComponentUsed',
+        text: `ModuleRegistry.component ${globalID} used but not yet registered`
+      });
```

Comment:
I think instead of types it's worth considering using custom error objects here, assuming:
```js
class UnregisteredComponentError extends Error {}
```

We could:
```js
this.notifyListeners('reactModuleContainer.error', new UnregisteredComponentError(`ModuleRegistry.component ${globalID} used but not yet registered`);
```

That is, basically use the native error object and mechanism instead of creating a new one. I'm not at all convinced it's the right way to go, but I think we should consider it.